aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -465,15 +465,20 @@ module aptos_framework::delegation_pool {
         assert!(delegation_pool_exists(pool_address), error::invalid_argument(EDELEGATION_POOL_DOES_NOT_EXIST));
     }
 
-    fun assert_min_active_balance(pool: &DelegationPool, delegator_address: address) {
+    fun assert_min_active_balance(pool: &DelegationPool, delegator_address: address, or_zero_balance: bool) {
+        let shares = pool_u64::shares(&pool.active_shares, delegator_address);
         let balance = pool_u64::balance(&pool.active_shares, delegator_address);
-        assert!(balance >= MIN_COINS_ON_SHARES_POOL, error::invalid_argument(EDELEGATOR_ACTIVE_BALANCE_TOO_LOW));
+        assert!(
+            balance >= MIN_COINS_ON_SHARES_POOL || (or_zero_balance && shares == 0),
```

Comment:
As discussed offline, this would be more readable IMO if or_zero_balance is checked in unlock.